src/update/UpdateManager.cpp

```diff
@@ -65,34 +65,43 @@ VersionPatch UpdateManager::GetVersionPatch() const
     return mVersionPatch;
 }
 
-void UpdateManager::GetUpdates()
+void UpdateManager::GetUpdates(bool ignoreNetworkErrors)
 {
     const audacity::network_manager::Request request("https://updates.audacityteam.org/feed/latest.xml");
     auto response = audacity::network_manager::NetworkManager::GetInstance().doGet(request);
 
-    response->setRequestFinishedCallback([response, this](audacity::network_manager::IResponse*) {
+    response->setRequestFinishedCallback([response, ignoreNetworkErrors, this](audacity::network_manager::IResponse*) {
 
         auto gAudioIO = AudioIO::Get();
         if (response->getError() != audacity::network_manager::NetworkError::NoError)
         {
-            gAudioIO->CallAfterRecording([] {ShowExceptionDialog(nullptr,
-                XC("Error checking for update", "update dialog"),
-                XC("Unable to connect to Audacity update server.", "update dialog"),
-                wxString());
-                });
-
-            return;
+           if (!ignoreNetworkErrors)
+           {
+              gAudioIO->CallAfterRecording([] {
+                 ShowExceptionDialog(
+                    nullptr, XC("Error checking for update", "update dialog"),
+                    XC("Unable to connect to Audacity update server.",
+                       "update dialog"),
+                    wxString());
+              });
+           }
+           
+           return;
         }
 
         if (!mUpdateDataParser.Parse(response->readAll<VersionPatch::UpdateDataFormat>(), &mVersionPatch))
         {
-            gAudioIO->CallAfterRecording([] {ShowExceptionDialog(nullptr,
-                XC("Error checking for update", "update dialog"),
-                XC("Update data was corrupted.", "update dialog"),
-                wxString());
-                });
-
-            return;
+           if (!ignoreNetworkErrors)
+           {
+              gAudioIO->CallAfterRecording([] {
```

Comment:
You would suppress this call too.  Not necessary to avoid the silliness of putting up an error, just because the user turned of wifi.  But it makes sense too, if perhaps you get a connection but then for some reason get garbage from it.  Don't try to send an error.